File-Formats/Presentation/Working-with-find-and-replace.md

```diff
@@ -0,0 +1,932 @@
+//Instantiates the File Picker
+FileOpenPicker openPicker = new FileOpenPicker();
+openPicker.SuggestedStartLocation = PickerLocationId.Desktop;
+openPicker.FileTypeFilter.Add(".pptx");
+//Creates a storage file from FileOpenPicker
+StorageFile inputStorageFile = await openPicker.PickSingleFileAsync();
+//Loads or open an PowerPoint Presentation
+IPresentation pptxDoc = await Presentation.OpenAsync(inputStorageFile);
+//Finds all the occurrences of a particular text in the PowerPoint presentation
+ITextSelection[] textSelections = pptxDoc.FindAll("product", false, false);
+foreach (ITextSelection textSelection in textSelections)
+{
+	//Gets the found text as single text part
+	ITextPart textPart = textSelection.GetAsOneTextPart();
+	//Replace the text
+	textPart.Text = "Service";
+}
+//Initializes FileSavePicker
+FileSavePicker savePicker = new FileSavePicker();
+savePicker.SuggestedStartLocation = PickerLocationId.Desktop;
+savePicker.SuggestedFileName = "Output";
+savePicker.FileTypeChoices.Add("PowerPoint Files", new List<string>() { ".pptx" });
+//Creates a storage file from FileSavePicker
+StorageFile storageFile = await savePicker.PickSaveFileAsync();
+//Saves changes to the specified storage file
+await pptxDoc.SaveAsync(storageFile);
+{% endhighlight %}
+
+{% highlight c# tabtitle="ASP.NET Core" %}
+//Opens an existing presentation.           
+FileStream inputStream = new FileStream("Sample.pptx", FileMode.Open);
+using (IPresentation pptxDoc = Presentation.Open(inputStream))
+{
+	//Finds all the occurrences of a particular text in the PowerPoint presentation
+	ITextSelection[] textSelections = pptxDoc.FindAll("product", false, false);
+	foreach (ITextSelection textSelection in textSelections)
+	{
+		//Gets the found text as single text part
+		ITextPart textPart = textSelection.GetAsOneTextPart();
+		//Replace the text
+		textPart.Text = "Service";
+	}
+	//Save the PowerPoint Presentation as stream
+	FileStream outputStream = new FileStream("Output.pptx", FileMode.Create);
+	pptxDoc.Save(outputStream);
+	//Release all resources of the stream
+	outputStream.Dispose();
+	//Closes the Presentation instance
+	pptxDoc.Close();
+}
+{% endhighlight %}
+
+{% highlight c# tabtitle="Xamarin" %}
+//"App" is the class of Portable project.
+Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+Stream inputStream = assembly.GetManifestResourceStream("SampleBrowser.Presentation.Samples.Template.Sample.pptx");
+//Loads or open an PowerPoint Presentation
+IPresentation pptxDoc = Presentation.Open(inputStream);
+//Finds all the occurrences of a particular text in the PowerPoint presentation
+ITextSelection[] textSelections = pptxDoc.FindAll("product", false, false);
+foreach (ITextSelection textSelection in textSelections)
+{
+	//Gets the found text as single text part
+	ITextPart textPart = textSelection.GetAsOneTextPart();
+	//Replace the text
+	textPart.Text = "Service";
+}
+//Create new memory stream to save Presentation.
```

Comment:
Create a new memory stream to save Presentation.

Reply:
Modified as you said.